### PR TITLE
Add mimetype module

### LINKED ISSUE #84: Start making WRY more DRY
If you are looking to get your hands dirty, this is a great issue to get started working with Rust and WRY. 

In this PR, https://github.com/tauri-apps/wry/pull/82 there is some code repetition, and if you look closely in the codebase, there are some other cases of it. With regard to mime types (aka media types), there may be situations where we need to add others as time goes on, so having the list in one place makes it easier to update and know that we have feature parity.

Feel free to ask questions here in the issue, or on our discord server.

### PR DESCRIPTION
**Pull request by Shriram-Balaji**
Fixes #84 

* Added a module to centralize `MimeType` parsing 
* Also added a debug configuration for using the debugger to easily walkthrough examples.
* Fixed failing doctests. 

  > NOTE: Some of the method signatures were slightly out of date. Although `no_run` flag is added in the comment, while running `cargo test` on nightly, the docs would not compile. However the docs remain unchanged for the most part, as all the code added is just to satisfy `cargo` from compiling the code.

### CODE REVIEW

**Review by nothingismagick — COMMENTED** [review]
Is the `.vscode/launch.json` needed?

**Comment by Shriram-Balaji** [comment]
Not really, but it was quite useful to debug the examples and see what was going on.

Can .gitignore if it feels unnecessary.

( FWIW the configurations were auto added by vscode )

File: src/lib.rs [hunk 1]
```diff
@@ -8,7 +8,11 @@
 //! unified APIs across all platforms. To get started, you simply create an [`Application`] first:
 //!
 //! ```no_run
+//! # use wry::Application;
+//! # fn main() -> Result<(), wry::Error> {
 //! let application = Application::new()?;
+//! # Ok(())
+//! # }
```
Comment (nklayman) [inline src/lib.rs:15]: Why were these added? They seem unrelated to this PR.

Comment (Shriram-Balaji) [reply src/lib.rs:15]: Tried running `cargo test` in the root, and got failing doctests. Thought I'd fix them while I was at it, but yeah they are unrelated. 

Maybe I should move these changes into a new PR if need be

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,212 @@
+{
```
Comment (wusyong) [inline .vscode/launch.json:1]: Please remove all configs under `.vscode/` and add a global gitignore to your environment.
So next time it won't be added no matter what gitignore file is in each individual project.

File: src/lib.rs [hunk 3]
```diff
@@ -17,6 +21,9 @@
 //! them with `Default::default()` and `None`.
 //!
 //! ```no_run
+//! # use wry::{Application, Attributes, Callback};
+//! # fn main() -> Result<(), wry::Error> {
+//! # let mut app = Application::new()?;
```
Comment (wusyong) [inline src/lib.rs:26]: There's a no run attribute. So you can remove these

**Review by wusyong — CHANGES_REQUESTED** [review]
PR looks good to go. Just some nits to be cleaned.

File: .vscode/launch.json (hunk 2 above)
Comment (wusyong) [reply .vscode/launch.json:1]: This [link](https://docs.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) might help you.

File: src/lib.rs (hunk 3 above)
Comment (Shriram-Balaji) [reply src/lib.rs:26]: Inspite of having a no_run the doc-test failed to compile for me, hence added it.

Comment (wusyong) [reply src/lib.rs:26]: how about change `no_run` to other attributes like `ignore` or `text`?

Comment (Shriram-Balaji) [reply src/lib.rs:26]: @wusyong `ignore` feels ok, but the docs could potentially become out-of-date. I'll remove this change from this PR for now, and maybe go over it in a separate one as it seems unrelated to #84

**Review by wusyong — APPROVED** [review]
